src/zh/yidan/src/eu/kanade/tachiyomi/extension/zh/yidan/Yidan.kt

```diff
@@ -1,135 +1,347 @@
+                categoryId = filters.firstInstance<CategoryFilter>().selected,
+                orderType = filters.firstInstance<SortFilter>().selected,
+                overType = filters.firstInstance<StatusFilter>().selected,
+            ).toJsonRequestBody(),
+        )
     }
 
-    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+    override fun searchMangaParse(response: Response): MangasPage {
+        val searchByKeyword = response.request.url.toString().contains("searchNovel")
+        val records = if (searchByKeyword) {
+            response.parseAs<CommonResponse<List<Record>>>().result
+        } else {
+            response.parseAs<CommonResponse<FilterResult>>().result.list
+        }
+        return createMangasPage(records, paginated = !searchByKeyword)
+    }
 
-    // for WebView
-    override fun mangaDetailsRequest(manga: SManga) =
-        GET("$baseUrl/#/pages/detail/detail?id=${manga.url}")
+    private fun createMangasPage(records: List<Record>, paginated: Boolean = true): MangasPage {
+        return MangasPage(
+            records.map {
+                SManga.create().apply {
+                    url = "${it.id}"
+                    title = it.novelTitle
+                    thumbnail_url = it.imgUrl
+                }
+            },
+            if (paginated) {
+                records.size >= PAGE_SIZE
+            } else {
+                false
+            },
+        )
+    }
 
-    override fun fetchMangaDetails(manga: SManga): Observable<SManga> {
-        val request = GET("$baseUrl/prod-api/app-api/vv/mh-list/get?id=${manga.url}", headers)
-        return client.newCall(request).asObservableSuccess().map { mangaDetailsParse(it) }
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/pages/comic/info?id=${manga.url}"
     }
 
-    override fun mangaDetailsParse(response: Response) =
-        response.parseAs<MangaDto>().toSManga(baseUrl)
+    override fun mangaDetailsRequest(manga: SManga) = chapterListRequest(manga)
 
-    override fun chapterListRequest(manga: SManga) =
-        GET("$baseUrl/prod-api/app-api/vv/mh-episodes/list?mhid=${manga.url}", headers)
+    override fun mangaDetailsParse(response: Response): SManga {
+        val comic = response.parseAs<CommonResponse<ComicInfoResult>>().result.comic
+        return SManga.create().apply {
+            url = "${comic.id}"
+            title = comic.novelTitle
+            thumbnail_url = comic.bigImgUrl
+            genre = comic.tags
+            author = comic.author
+            description = comic.introduction
+            status = when (comic.overType) {
+                1 -> SManga.ONGOING
+                2 -> SManga.COMPLETED
+                else -> SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return "$baseUrl/pages/comic/content?f=1&s=${chapter.chapter_number.toInt()}"
```

Comment:
Consider:
```suggestion
        return "$baseUrl/pages/comic/content.toHttpUrl().newBuilder()
            .addQueryParameter("f", "1")
            .addQueryParameter("s", chapter.chapter_number.toInt().toString())
            .toString()
```
whitespace might be off.